test/integration/urdf_to_sdf.cc

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <string>
+#include <gtest/gtest.h>
+
+#include "sdf/sdf.hh"
+#include "sdf/parser.hh"
+#include "sdf/ParserConfig.hh"
+#include "test_config.h"
+
+/////////////////////////////////////////////////
+sdf::SDFPtr InitSDF()
+{
+  sdf::SDFPtr sdf(new sdf::SDF());
+  sdf::init(sdf);
+  return sdf;
+}
+
+/////////////////////////////////////////////////
+/// Check that _a contains _b
+static bool contains(const std::string &_a, const std::string &_b)
+{
+  return _a.find(_b) != std::string::npos;
+}
+
+//////////////////////////////////////////////////
+TEST(URDF2SDF, ValidBasicURDF)
+{
+  std::string urdfXml = R"(
+    <robot name='test_robot'>
+      <link name='link1'>
+        <inertial>
+          <mass value='0.1' />
+          <origin rpy='1.570796326794895 0 0' xyz='0.123456789123456 0 0.0' />
+          <inertia ixx='0.01' ixy='0' ixz='0' iyy='0.01' iyz='0' izz='0.01' />
+        </inertial>
+      </link>
+    </robot>)";
+
+  auto sdf = InitSDF();
+  sdf::ParserConfig config;
+  sdf::Errors errors;
+
+  ASSERT_TRUE(sdf::readString(urdfXml, config, sdf, errors));
+  ASSERT_TRUE(errors.empty()) << errors;
+  ASSERT_TRUE(sdf);
+
+  sdf::Root root;
+  errors = root.Load(sdf);
```

Comment:
You should be able to just load `urdfXml` using `LoadSdfString`. No need for `readString`.

Reply:
Yeah I noticed that, but I was wondering if that might introduce regressions in the future, if we ever changed `LoadSdfString` to only load sdf, and not `urdf`, so I figured I should be more explicit. Let me know if the behavior change is highly unlikely, I would refactor this portion then :laughing:

Reply:
There is no plan to make that change right now.  I think we should always make sure to test the current API regardless of future plans, especially since  `Root::Load` and `Root::LoadSdfString` are the main entry points for most users.

Reply:
https://github.com/gazebosim/sdformat/pull/1238/commits/ecdad415e2794b031039afb5b850d22c4066b18c